config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml

```diff
@@ -187,6 +187,25 @@ spec:
                 required:
                 - type
                 type: object
+              imageLookupBaseOS:
```

Comment:
wdyt moving it under a container so that it can be extended in future?
```yaml
imageLookup:
  baseOS:
  format:
```

Reply:
i used the same convention as CAPA and other providers which has them on the top level.

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/07a93a51edacdf97c326c8e5f3016b41be6bab40/api/v1beta2/awsmachine_types.go#L84

this is also used by tinkerbell 

https://github.com/tinkerbell/cluster-api-provider-tinkerbell/blob/main/api/v1beta1/tinkerbellmachine_types.go#L49

Reply:
We should have a reason for diverging from the convention, and I think @supershal gives a good one.

(As an aside, that tinkerbell change was authored by a former CAPA maintainer 4 years ago.)